compiler/next/include/chpl/queries/Context.h

```diff
@@ -346,6 +371,49 @@ class Context {
    */
   void setFilePathForModuleID(ID moduleID, UniqueString path);
 
+  /**
+    Note an error for the currently running query.
+   */
+  void error(ErrorMessage error);
```

Comment:
Is this function supposed to be used to only report queries? Maybe name queryError()? Context isn't exclusively used for error reporting right? Given that, I might assume this is a general "error out during compilation" mechanism.

Reply:
When using the query framework, this is what should be used to report an error. We shouldn't have any "error out during compilation" calls because if they are there, they mess up the goals of the compiler library and query framework. So I chose the shortest name for this one since it is the one code should be using.

Reply:
Makes sense. I assume there's some mechanism we have for adding an error message to a log of errors (that we attempt to recover from to continue compilation and finding further errors). My concern would be if this function would easily get confused for that.